src/Meziantou.Analyzer/Rules/AvoidUsingRedundantElseFixer.cs

```diff
@@ -38,75 +38,46 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
 
         private static async Task<Document> RemoveRedundantElse(Document document, SyntaxNode nodeToFix, CancellationToken cancellationToken)
         {
-            var elseClause = (ElseClauseSyntax)nodeToFix;
-            if (elseClause == null)
+            if (!(nodeToFix is ElseClauseSyntax elseClauseSyntax))
                 return document;
 
-            var ifStatement = elseClause.Parent as IfStatementSyntax;
-            if (ifStatement == null)
+            if (!(elseClauseSyntax.Parent is IfStatementSyntax ifStatementSyntax))
                 return document;
 
-            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var ifStatementParent = ifStatementSyntax.Parent;
+            if (ifStatementParent is null)
+                return document;
 
-            // Remove else
-            root = root.ReplaceNode(ifStatement, ifStatement.WithElse(@else: null));
-            ifStatement = (IfStatementSyntax)root.FindNode(ifStatement.IfKeyword.Span);
+            var modifiedIfStatement = ifStatementSyntax
+                .WithElse(null)
+                .WithTrailingTrivia(ifStatementSyntax.GetTrailingTrivia().Add(LineFeed));
```

Comment:
Could you add a test with an empty else statement

````
void A()
{
    if (true)
    {
         return;
    }
    else
    {
    }
}
````

Should fix with the following code (no empty line after the if)

````
void A()
{
    if (true)
    {
         return;
    }
}
````

Reply:
Good idea! It turns out the fixer was still outputting the empty line after the `if` block. 🙄 
Fixing it...